package/MDAnalysis/analysis/contacts.py

```diff
@@ -773,3 +773,152 @@ def plot_qavg(self, filename=None, **kwargs):
+
+    """    
+    # reference groups A and B from selection strings
+    refA, refB = ref.select_atoms(selA), ref.select_atoms(selB)
+
+    # 2D float array, reference distances (r0)
+    dref = distance_array(refA.coordinates(), refB.coordinates())
+
+    # 2D bool array, select reference distances that are less than the cutoff radius
+    mask = dref < radius
+    #print("ref has {:d} contacts within {:.2f}".format(mask.sum(), radius))
+
+    # group A and B in a trajectory
+    grA, grB = u.select_atoms(selA), u.select_atoms(selB)
+    results = []
+
+
+    for ts in u.trajectory:
+        d = distance_array(grA.coordinates(), grB.coordinates())
+        r, r0 = d[mask], dref[mask]
+        x = 1/(1 + np.exp(beta*(r - alpha * r0)))
+        print x
+        results.append(( ts.time, x.sum()/mask.sum() ))
+    #results = pd.DataFrame(results, columns=["Time (ps)", "Q"])
+    return results
+
+from MDAnalysis.analysis.distances import distance_array
+from .base import AnalysisBase
+logger = logging.getLogger(__name__)
+
+class BestHummerContacts(AnalysisBase):
+    r"""Calculate the persistence length for polymer chains
+
+    The persistence length is the length at which two points on the polymer
+    chain become decorrelated.
+
+    Notes
+    -----
+    This analysis requires that the trajectory supports indexing
+
+    .. versionadded:: 0.13.0
+    """
+    def __init__(self, u, grA, grB, refA, refB, radius=4.5, lambda_constant=1.8, beta=5.0,
+                 start=None, stop=None, step=None):
+        """Calculate the persistence length for polymer chains
+
+        Parameters
+        ----------
+        u : Universe
+            the thing with frame
+        grA, grB:
+            two contacting groups that change over time
+        refA, refB:
+            two contacting groups in their reference conformation
+        radius:
+            radius within which contacts exist
+        lambda_constant: 
+            contact is considered formed between (lambda*r0,r0)
+        beta:
+            softness of the switching function, the lower the softer
+        start : int, optional
+            First frame of trajectory to analyse, Default: 0
+        stop : int, optional
+            Last frame of trajectory to analyse, Default: -1
+        step : int, optional
+            Step between frames to analyse, Default: 1
+        """
+        self.u = u
+        self.grA, self.grB = grA, grB
+
+        r0 = distance_array(refA.coordinates(), refB.coordinates())
+        self.r0 = r0
+        self.mask = r0 < radius
+        self.results = []
+
+        self.beta = beta
+        self.lambda0  = lambda_constant
+
+        self._setup_frames(self.u.trajectory,
+                           start=start,
+                           stop=stop,
+                           step=step)
+
+    def _single_frame(self):
+        grA, grB, r0, mask = self.grA, self.grB, self.r0, self.mask
+
+        d = distance_array(grA.coordinates(), grB.coordinates())
```

Comment:
I don't know if it's actually deprecated? But i think  we're trying to use '.positions' rather than coordinates()